### PR TITLE
Jobradovic points component

### PR DESCRIPTION
**Pull request by jolpazy**

### CODE REVIEW

File: web/src/main/frontend/src/base/stores/CompletedTestStore.js [hunk 1]
```diff
@@ -57,17 +57,30 @@ class CompletedTest {
...
+      availablePoints: 40
     }
   ];
+
+  @observable scores = {};
+
+  @observable totalScore = 0;
+
+  @observable
+  perfectScore = this.questions.reduce((a, b) => a + b.availablePoints, 0);
```
Comment (iscekic) [inline web/src/main/frontend/src/base/stores/CompletedTestStore.js]: Can this be `computed`?

File: web/src/main/frontend/src/base/stores/CompletedTestStore.js [hunk 2]
```diff
@@ -57,17 +57,30 @@ class CompletedTest {
...
+
+  @action
+  addScore = (value, key) => {
+    this.scores[key] = value;
+    this.totalScore = Object.values(this.scores).reduce((a, b) => a + b, 0);
```
Comment (iscekic) [inline web/src/main/frontend/src/base/stores/CompletedTestStore.js]: `Object.values` isn't supported in IE11. I can't remember if we support it. 😄 If we do, please use `R.values`.

File: web/src/main/frontend/src/components/Points.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import {inject} from 'mobx-react';
+
+@inject('completedTest')
+class Points extends React.Component {
+  id = this.props.id;
```
Comment (iscekic) [inline web/src/main/frontend/src/components/Points.js]: Why is this here?

File: web/src/main/frontend/src/components/Points.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+      this.props.completedTest.addScore(this.props.points, this.props.id);
+    }
+  }
+
+  wheel = e => e.preventDefault();
+
+  onChange = event => {
+    let value = parseInt(event.target.value, 10);
+    this.props.completedTest.addScore(value, this.props.id);
+  };
```
Comment (iscekic) [inline web/src/main/frontend/src/components/Points.js]: Missing new line below.

File: web/src/main/frontend/src/components/Points.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  render() {
+    return (
+      <div className="d-flex justify-content-end">
+        {this.props.fixedScore ? (
```
Comment (iscekic) [inline web/src/main/frontend/src/components/Points.js]: This double ternary should be extracted into a method.

File: web/src/main/frontend/src/components/Points.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+@inject('completedTest')
+class Points extends React.Component {
+  id = this.props.id;
+  componentDidMount() {
+    if (this.props.isCorrect) {
+      this.props.completedTest.addScore(this.props.points, this.props.id);
+    }
+  }
+
+  wheel = e => e.preventDefault();
```
Comment (iscekic) [inline web/src/main/frontend/src/components/Points.js]: Rename to `turnOffScrollWheel` or similar. 😄

File: web/src/main/frontend/src/components/Points.js [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+          <span>
+            <input
+              // Turning off mouseWheel event on number input
+              onWheel={e => this.wheel(e)}
+              className="noSpinner"
+              min={0}
+              max={this.props.points}
+              type="number"
+              onChange={this.onChange}
+              name=""
```
Comment (iscekic) [inline web/src/main/frontend/src/components/Points.js]: Do we need `name` and `id`?

File: web/src/main/frontend/src/scss/components/_reviewTest.scss [hunk 8]
```diff
@@ -66,9 +66,34 @@ textarea {
...
   height: 170px !important;
+  margin-bottom: 8px !important;
+}
+
+.noSpinner::-webkit-outer-spin-button,
+.noSpinner::-webkit-inner-spin-button {
+  appearance: none;
+}
+
+.noSpinner {
```
Comment (iscekic) [inline web/src/main/frontend/src/scss/components/_reviewTest.scss]: Class names should dashes.

File: web/src/main/frontend/src/scss/components/_reviewTest.scss [hunk 9]
```diff
@@ -66,9 +66,34 @@ textarea {
...
 }
 
 // !important is the only way to style CodeMirror component other than editing css in node_modules
 .CodeMirror {
   height: 170px !important;
+  margin-bottom: 8px !important;
+}
+
+.noSpinner::-webkit-outer-spin-button,
+.noSpinner::-webkit-inner-spin-button {
```
Comment (iscekic) [inline web/src/main/frontend/src/scss/components/_reviewTest.scss]: Use `&`. https://css-tricks.com/the-sass-ampersand/

File: web/src/main/frontend/src/base/stores/CompletedTestStore.js (hunk 1 above)
Comment (jolpazy) [reply web/src/main/frontend/src/base/stores/CompletedTestStore.js]: Sure

File: web/src/main/frontend/src/components/Points.js (hunk 3 above)
Comment (jolpazy) [reply web/src/main/frontend/src/components/Points.js]: it's a leftover (:

File: web/src/main/frontend/src/components/Points.js (hunk 7 above)
Comment (jolpazy) [reply web/src/main/frontend/src/components/Points.js]: We most certainly don't (:

File: web/src/main/frontend/src/base/stores/CompletedTestStore.js (hunk 2 above)
Comment (iscekic) [reply web/src/main/frontend/src/base/stores/CompletedTestStore.js]: Yes, we included `Ramda` a week or so ago. 😄

Comment (iscekic) [reply web/src/main/frontend/src/base/stores/CompletedTestStore.js]: Meh, screw IE, leave it like this. 😄

File: web/src/main/frontend/src/components/Points.js [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+@inject('completedTest')
+class Points extends React.Component {
+  componentDidMount() {
+    if (this.props.isCorrect) {
+      this.props.completedTest.addScore(this.props.points, this.props.id);
+    }
+  }
+
+  createInputField(fixed, correct, valuated) {
+    if (fixed) {
```
Comment (iscekic) [inline web/src/main/frontend/src/components/Points.js]: I dislike this. Can it look something like:

```js
createInputField(fixed, correct, valuated) {
  const className = // ... get class name for given params
  const handler = // ... get handler for given params
  return (
    <input
      // ...
    />
  );
}
```

Comment (iscekic) [reply web/src/main/frontend/src/components/Points.js]: You can use `() => {}` or http://ramdajs.com/docs/#always when you need no-op. :smile:

File: web/src/main/frontend/src/components/Points.js [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+    }
+  }
+
+  turnOffScrollWheell = e => e.preventDefault();
+
+  onChange = event => {
+    let value = parseInt(event.target.value, 10);
+    if (isNaN(value) || value <= 0) {
+      value = 0;
+      event.target.value = null;
```
Comment (iscekic) [inline web/src/main/frontend/src/components/Points.js]: Why are we changing `event.target.value` directly?

File: web/src/main/frontend/src/components/Points.js [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+          placeholder={this.props.points}
+        />
+      );
+    }
+  }
+
+  turnOffScrollWheell = e => e.preventDefault();
+
+  onChange = event => {
+    let value = parseInt(event.target.value, 10);
```
Comment (iscekic) [inline web/src/main/frontend/src/components/Points.js]: Add some newlines. 😄

File: web/src/main/frontend/src/scss/components/_reviewTest.scss [hunk 13]
```diff
@@ -66,9 +66,36 @@ textarea {
...
 // !important is the only way to style CodeMirror component other than editing css in node_modules
 .CodeMirror {
   height: 170px !important;
+  margin-bottom: 8px !important;
+}
+
+.no-spinner {
+  text-align: center;
+  width: 60px;
+  &::-webkit-inner-spin-button,
```
Comment (iscekic) [inline web/src/main/frontend/src/scss/components/_reviewTest.scss]: `/n` please. 😄

File: web/src/main/frontend/src/scss/components/_reviewTest.scss [hunk 14]
```diff
@@ -66,9 +66,36 @@ textarea {
...
+.wrong {
+  //placeholder color:
+  color: rgb(245, 131, 131);
+}
+
+.result {
+  width: 100px;
+  text-align: center;
+  margin-left: 4px;
+}
```
Comment (iscekic) [inline web/src/main/frontend/src/scss/components/_reviewTest.scss]: `/n` please. 😄

File: web/src/main/frontend/src/components/Points.js [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+          className="no-spinner"
+          type="number"
+          onChange={this.onChange}
+          placeholder={this.props.points}
+        />
+      );
+    }
+  }
+
+  turnOffScrollWheell = e => e.preventDefault();
```
Comment (iscekic) [inline web/src/main/frontend/src/components/Points.js]: Typo here, should be `wheel`

File: web/src/main/frontend/src/components/Points.js (hunk 11 above)
Comment (jolpazy) [reply web/src/main/frontend/src/components/Points.js]: so it falls back to 0 as a value, and placeholder in the view
